File: problems/2395-find-subarrays-with-equal-sum/solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean findSubarrays(int[] nums) {
        Set<Integer> sums = new HashSet<>();
        for (int i = 0; i < nums.length - 1; i++) {
            int sum = nums[i] + nums[i + 1];
            if (sums.contains(sum)) {
                return true;
            }
            sums.add(sum);
        }
        return false;
    }
}

File: problems/2395-find-subarrays-with-equal-sum/solution.c
bool findSubarrays(int* nums, int numsSize) {
    if (numsSize < 2) {
        return false;
    }

    int sums[1000];
    for (int i = 0; i < numsSize - 1; i++) {
        sums[i] = nums[i] + nums[i + 1];
    }

    for (int i = 0; i < numsSize - 1; i++) {
        for (int j = i + 1; j < numsSize - 1; j++) {
            if (sums[i] == sums[j]) {
                return true;
            }
        }
    }

    return false;
}